src/dao/LOCsManagerDAO.js

```diff
@@ -87,39 +87,39 @@ class LOCsManagerDAO extends AbstractContractDAO {
         return this.contract.then(deployed => deployed.removeLOC(address, {from: account, gas: 3000000}));
     };
 
-    newLOCWatch = callback => this.contract.then(deployed => {
+    newLOCWatch = (callback, account: string) => this.contract.then(deployed => {
         const blockNumber = this.web3.eth.blockNumber;
-        deployed.newLOC({}, {}, (e, r) => {
+        this._watch(deployed.newLOC, (r, block, ts) => {
             if (r.blockNumber <= blockNumber) return;
```

Comment:
1. r.blockNumber === block
2. This logic is wrong. You should treat all events, that _watch will give you, so user will have all notices. This is exactly why we've implemented _watch.